src/DictVectors/pdvec.jl

```diff
@@ -567,6 +582,65 @@ function Base.map!(f, dst::PDVec, src::PDVecVals)
     return dst
 end
 
+"""
+    map(f, values(::PDVec))
+
+Out-of-place parallel `map` on values of a [`PDVec`](@ref). Returns a new
+[`PDVec`](@ref). Only defined for `values` as efficiently changing keys in a thread-safe and
+distributed way is not possible.
+"""
+function Base.map(f, src::PDVecVals)
+    return map!(f, copy(src.vector), src)
+end
+
+"""
+    filter!(pred, [dst::PDVec, ], keys(::PDVec))
+    filter!(pred, [dst::PDVec, ], values(::PDVec))
+    filter!(pred, [dst::PDVec, ], pairs(::PDVec))
+
+In-place parallel `filter!` on an iterator over a [`PDVec`](@ref). If `dst` is provided,
+results are written there.
+"""
+function Base.filter!(f::F, src::PDVecIterator) where {F}
+    Folds.foreach(src.vector.segments) do segment
+        if src.selector === pairs
+            filter!(f, segment)
+        elseif src.selector === keys
+            filter!((k, _) -> f(k), segment)
+        elseif src.selector === values
+            filter!((_, v) -> f(v), segment)
+        end
+    end
+    return src.vector
+end
+
+function Base.filter!(f::F, dst::PDVec, src::PDVecIterator) where {F}
+    if dst === src.vector
+        return filter!(f, src::PDVecIterator)
```

Comment:
Perhaps I am misunderstanding, but the out-of-place method `filter` which copies the vector should then call `filter!` here, but I don't think it does? So the error I mentioned above does not appear when using the out of place method

Reply:
It is not covered in the tests

Reply:
That's right, fixed.